web/src/components/game/goals/GoalManagement.tsx

```diff
@@ -149,6 +196,38 @@ export default function GoalManagement() {
                     />
                 </DialogContent>
             </Dialog>
+
+            {/* Delete Confirmation Dialog */}
+            <Dialog ref={deleteDialogRef}>
```

Comment:
If we're going to seat multiple dialogs into a single component we should either
a) make them each their own component, and utilize `useImperativeHandle`/`forwardRef` to isolate state/functionality
b) Use a modular composing system for the content so we only seat a single dialog and have a single ref, such as something like the following
```tsx
const [dialogContent, setDialogContent] = useState(null)
const launchDelateAllConfirmation = () => {
    setDialogContent(<DeleteAllConfirmationDialog />)
    dialogRef.current.open()
}
...
return (
    ...
    <Button onClick={launchDeleteAllConfirmation}
    <Dialog ref={dialogRef}>
        {dialogContent}
    </Dialog>
)
```